### PR TITLE
[luci-interpreter] Support ReduceMax Op

### PR DESCRIPTION
**Pull request by jinevening**
This supports ReduceMax Op execution.

ONE-DCO-1.0-Signed-off-by: Hyukjin Jeong <hj1.jeong@samsung.com>

---
Related to: https://github.com/Samsung/ONE/issues/9458

### CODE REVIEW

File: compiler/luci-interpreter/src/kernels/ReduceMax.h [hunk 1]
```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright (c) 2020 Samsung Electronics Co., Ltd. All Rights Reserved
```
Comment (seanshpark) [inline compiler/luci-interpreter/src/kernels/ReduceMax.h:2]: 2022?

**Review by seanshpark — APPROVED** [review]
LGTM

File: compiler/luci-interpreter/src/kernels/ReduceMax.test.cpp [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+  std::unique_ptr<IMemoryManager> _memory_manager;
+};
+
+TEST_F(ReduceMaxTest, FloatNotKeepDims)
+{
+  std::vector<float> input_data = {1.0,  2.0,  3.0,  4.0,  5.0,  6.0,  7.0,  8.0,
+                                   9.0,  10.0, 11.0, 12.0, 13.0, 14.0, 15.0, 16.0,
+                                   17.0, 18.0, 19.0, 20.0, 21.0, 22.0, 23.0, 24.0};
+
+  std::vector<int32_t> axis_data{1, 0, -3, -3};
```
Comment (llFreetimell) [inline compiler/luci-interpreter/src/kernels/ReduceMax.test.cpp:45]: input is rank-3 but axis is rank-4.. is it possible?

Comment (jinevening) [reply compiler/luci-interpreter/src/kernels/ReduceMax.test.cpp:45]: Yes. I brought this case from tflite test codes.

Comment (llFreetimell) [reply compiler/luci-interpreter/src/kernels/ReduceMax.test.cpp:45]: I see it, maybe it is interpreted as `{1, 0, -3, -3} = {1, 0, 0, 0} = {1, 0}` :)

Adding Tensorflow Copyright was good but adding reference link would be perfect..! :D
(You can add it later, not in this PR)

**Review by llFreetimell — APPROVED** [review]
LGTM

**Pull request merged**